### PR TITLE
Fix property caching

### PR DESCRIPTION
**Pull request by elandau**
Fix DefaultPropertyContainer so that it actually caches the new value and version stamp

### CODE REVIEW

File: archaius2-core/src/test/java/com/netflix/archaius/DefaultPropertyContainerTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    @Test
+    public void basicTest() {
+        SettableConfig config = new DefaultSettableConfig();
+        DefaultPropertyFactory factory = DefaultPropertyFactory.from(config);
+        Property<String> prop = factory.getProperty("foo").asString("default");
+
+        Assert.assertEquals("default", prop.get());
+        config.setProperty("foo", "value1");
+        Assert.assertEquals("value1", prop.get());
+        Assert.assertEquals("value1", prop.get());
```
Comment (howardyuan) [inline archaius2-core/src/test/java/com/netflix/archaius/DefaultPropertyContainerTest.java]: How do we know the second one uses cache and the first one not?

File: archaius2-core/src/main/java/com/netflix/archaius/property/DefaultPropertyContainer.java [hunk 2]
```diff
@@ -174,35 +174,6 @@ public void removeListener(PropertyListener<T> listener) {
...
-         * 
-         * @param latestVersion
-         * @return
-         */
-        T updateAndGet(int latestVersion) {
-            int[] cacheVersion = new int[1];
-            T existing = cache.get(cacheVersion);
-            try {
-                T value = resolveCurrent();
-                if (value != existing && (value == null || existing == null || !existing.equals(value))) {
```
Comment (howardyuan) [inline archaius2-core/src/main/java/com/netflix/archaius/property/DefaultPropertyContainer.java]: so the bug here is that if values are the same before current and cached, the version will not be updated?

File: archaius2-core/src/test/java/com/netflix/archaius/DefaultPropertyContainerTest.java (hunk 1 above)
Comment (elandau) [inline archaius2-core/src/test/java/com/netflix/archaius/DefaultPropertyContainerTest.java]: We currently don't.  The API would need to be re-worked to allow visibility into that.  For now I just stepped through it :(

File: archaius2-core/src/main/java/com/netflix/archaius/property/DefaultPropertyContainer.java (hunk 2 above)
Comment (elandau) [inline archaius2-core/src/main/java/com/netflix/archaius/property/DefaultPropertyContainer.java]: There were a few other use cases but essentially yes.  If the value was anything but new the version timestamp never got updated.

**Pull request merged**